src/carma-streets/parse_kafka_logs.py

```diff
@@ -45,20 +48,24 @@ def parse_kafka_logs_as_type(input_file_path: Path, message_type: KafkaLogMessag
         with open(input_file_path, encoding='utf8', errors='ignore') as log_file:
             msgs = []
             skipped_messages = 0
-            for line in log_file:
+            kafka_message = str()
+            for line_number, line in enumerate(log_file):
                 try:
                     line = line.strip()
                     #get the create time stamped by kafka
                     create_index = line.find('CreateTime')
                     if (create_index != -1):
+                        if kafka_message:
+                            kafka_json = json.loads(kafka_message)
+                            kafka_message = ''
+                            msgs.append(KafkaLogMessage(create_time, kafka_json, message_type))
                         create_time = re.sub('[^0-9]', '', line.split(':')[1])      
-
-                    json_beg_index = line.find('{')
-                    kafka_message = line[json_beg_index:]
-                    kafka_json = json.loads(kafka_message)
-                    msgs.append(KafkaLogMessage(create_time, kafka_json, message_type))
+                        json_beg_index = line.find('{')
+                        kafka_message += line[json_beg_index:]
+                    else:
+                        kafka_message += line
                 except json.JSONDecodeError as e:
-                    print(f'Error {e} extracting json info for message: {kafka_message}. Skipping message.')
+                    print(f'Error {e.msg} extracting json info for message: {kafka_message}. Skipping message.')
```

Comment:
The error message does not fit into the printout's sentence structure, and printing out the whole JSON payload may clutter the terminal. Consider an alternative structure:

```python
print(f"Skipping message: error parsing json info: {e}")
```

Reply:
```
Error Expecting property name enclosed in double quotes extracting json info for message: {     {"time_stamp":0,"name":"","intersections":[{"name":"Town04 Intersection 785","id":36243,"revision":1,"status":0,"moy":0,"time_stamp":20500,"states":[{"movement_name":"","signal_group":8,"state_time_speed":[{"event_state":8,"timing":{"start_time":205,"min_end_time":234,"max_end_time":234,"confidence":0}}]},{"movement_name":"","signal_group":7,"state_time_speed":[{"event_state":3,"timing":{"start_time":205,"min_end_time":205,"max_end_time":205,"confidence":0}}]},{"movement_name":"","signal_group":6,"state_time_speed":[{"event_state":3,"timing":{"start_time":205,"min_end_time":244,"max_end_time":244,"confidence":0}}]},{"movement_name":"","signal_group":5,"state_time_speed":[{"event_state":3,"timing":{"start_time":205,"min_end_time":205,"max_end_time":205,"confidence":0}}]},{"movement_name":"","signal_group":4,"state_time_speed":[{"event_state":8,"timing":{"start_time":205,"min_end_time":234,"max_end_time":234,"confidence":0}}]},{"movement_name":"","signal_group":3,"state_time_speed":[{"event_state":3,"timing":{"start_time":205,"min_end_time":205,"max_end_time":205,"confidence":0}}]},{"movement_name":"","signal_group":2,"state_time_speed":[{"event_state":3,"timing":{"start_time":205,"min_end_time":244,"max_end_time":244,"confidence":0}}]},{"movement_name":"","signal_group":1,"state_time_speed":[{"event_state":3,"timing":{"start_time":205,"min_end_time":205,"max_end_time":205,"confidence":0}}]}]}]}. Skipping message.
```
Below is example output from the script with this logging. You think there is something wrong with logging like this?

Reply:
I think using _action taken_, _reason why_, _other info_ makes it easier to parse at a glance. Thoughts?